### PR TITLE
(v4) Class + color generation, appearance theme propagation and more elements

### PR DESCRIPTION
**Pull request by nikosdouvlis**
## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend-core`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/edge`
- [ ] `build/tooling/chore`

## Description

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.

### CODE REVIEW

File: packages/clerk-js/src/v4/customizables/useContainerMetadata.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+
+type ContainerMetadataProps = {
+  flow: 'signIn' | 'signUp';
+  page: 'start' | 'emailOtp' | 'phoneOtp';
+  blockingAction?: boolean;
+};
+
+type ContainerMetadataContextValue = ContainerMetadataProps;
+const ContainerMetadataContext = React.createContext<{ value: ContainerMetadataContextValue } | undefined>(undefined);
```
Comment (raptisj) [inline packages/clerk-js/src/v4/customizables/useContainerMetadata.tsx:10]: nit: why is this type have to be reassigned? 
couldn't it be just this?

```
type ContainerMetadataContextValue = {
  flow: 'signIn' | 'signUp';
  page: 'start' | 'emailOtp' | 'phoneOtp';
  blockingAction?: boolean;
};
```

Comment (nikosdouvlis) [reply packages/clerk-js/src/v4/customizables/useContainerMetadata.tsx:10]: Thanks John, this has been fixed here #268 
Please do nit pick :)

File: packages/clerk-js/src/v4/customizables/makeCustomizable.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { useContainerMetadata } from '../elements/hooks';
+
+type Customizable<T = {}> = { appearanceKey?: string | string[] } & T;
+type CustomizablePrimitive<T> = React.FunctionComponent<Customizable<T>>;
+
+const fakeGenerateSemanticClassName = (props: any) => {
```
Comment (SokratisVidros) [inline packages/clerk-js/src/v4/customizables/makeCustomizable.tsx:6]: ❓ Why is this called `fakeGenerateSemanticClassName`? In other words, what's fake about it?

**Review by SokratisVidros — APPROVED** [review]
💯

File: packages/clerk-js/src/v4/customizables/makeCustomizable.tsx (hunk 2 above)
Comment (nikosdouvlis) [reply packages/clerk-js/src/v4/customizables/makeCustomizable.tsx]: (already discussed offline, posting here for the sake of completeness and resolving this one  :) )

`fakeGenerateSemanticClassName` was a placeholder for the actual className generation helpers which are now merged in this PR

**Comment by SokratisVidros** [comment]
@nikosdouvlis What's the status of this? Shall we merge?

**Comment by nikosdouvlis** [comment]
Closing this in favor of the `next` branch

**Pull request closed without merging**